### PR TITLE
[fix][client-producer] tweak the timeout msg before opMsg flush.

### LINKED ISSUE #15018: Client produce timeoutException error msg tweak.
**Describe the bug**
When ProducerImpl send msg timeout before org.apache.pulsar.client.impl.ProducerImpl.OpSendMsg#updateSentTimestamp, the error msg is not match the fact.

**Detail info**
When `OpSendMsg` before flush, the `firstSentAt` and `lastSentAt` would be `-1`.
So OpSendMsg sendComplete with TimeoutException, the errMsg's calculated logic would lead `firstSentAt` and `lastSentAt` not match the fact. 

see additional context.

**Additional context**
```
org.apache.pulsar.client.api.PulsarClientException$TimeoutException: The producer horizon-cluster-59-4 can not send message to the topic demo-topic within given timeout : createdAt 32.245 seconds ago, firstSentAt 80968.783 seconds ago, lastSentAt 80968.783 seconds ago, retryCount 0
	at org.apache.pulsar.client.api.PulsarClientException.unwrap(PulsarClientException.java:1008)
	at org.apache.pulsar.client.impl.TypedMessageBuilderImpl.send(TypedMessageBuilderImpl.java:89)
	at org.apache.pulsar.client.impl.ProducerBase.send(ProducerBase.java:62)
	at org.example.Producer.send(Producer.java:30)
	at org.example.Producer.main(Producer.java:12)
```

### PR DESCRIPTION
**Pull request by horizonzy**
### Motivation
Fixes #15018 

### Modifications
tweak the timeout msg at OpSendMsg.

### Documentation
 
- [x] `no-need-doc`

### CODE REVIEW

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java [hunk 1]
```diff
@@ -1408,9 +1408,11 @@ void sendComplete(final Exception e) {
                         te.getMessage(),
                         RelativeTimeUtil.nsToSeconds(ns - this.createdAt),
                         RelativeTimeUtil.nsToSeconds(this.firstSentAt <= 0
-                                ? ns - this.lastSentAt
```
Comment (HQebupt) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java:1411]: It is interesting. Do you mean setting it to -1 as `firstSendAt` value if `firstSendAt <= 0`?

Comment (horizonzy) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java:1411]: yep. -1 represents the msg not flush.

**Review by RobertIndie — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Review by hangc0276 — APPROVED** [review]
LGTM. Please add a doc to explain the -1 case.

**Comment by horizonzy** [comment]
> LGTM. Please add a doc to explain the -1 case.

done.

**Review by Jason918 — APPROVED** [review]

**Pull request merged**